include/macros.h

```diff
@@ -141,6 +141,13 @@
     }                                      \
     (void)0
 
+// horizontal, left to right
+#define PROJECTED_TO_SCREEN_X(projectedPos, invW) (s32)((projectedPos).x * (invW) * (SCREEN_WIDTH / 2) + (SCREEN_WIDTH / 2))
+// vertical, top to bottom
+#define PROJECTED_TO_SCREEN_Y(projectedPos, invW) (s32)((projectedPos).y * (invW) * (-SCREEN_HEIGHT / 2) + (SCREEN_HEIGHT / 2))
+// depth
+#define PROJECTED_TO_SCREEN_Z(projectedPos, invW) (s32)(((projectedPos).z * (invW)) * 16352) + 16352
```

Comment:
On what this does iirc we figured that out during environment system docs, maybe look for tharo's or my messages mentioning zbuffer in discord. I'll take a look too later

Reply:
Thanks past me because I completely forgot where this came from
https://github.com/zeldaret/oot/pull/1115#discussion_r789670711